Delete job prompt appears twice when deleting from table view

<!--
  Before opening a new issue, please search our existing issues: https://github.com/zowe/zowe-explorer-vscode/issues
-->

**Describe the bug**

If I open jobs in a table, then try to delete a job, I get the first prompt to delete the job. If I select `Delete`, the job is deleted, however I get a second (identical) prompt immediately after I hit `Delete` in the first prompt. Sometimes I get prompted again. If I click `Delete` in the subsequent prompts, I receive an error.

From logs:
```
[2025/04/28 09:19:49] [TRACE] Profiles.getInstance called.
[2025/04/28 09:19:49] [TRACE] JobTree.delete called.
[2025/04/28 09:19:49] [ERROR] EntryNotFound (FileSystemError): zowe-jobs:/demo.zosmf/JOB08156
[2025/04/28 09:19:49] [ERROR] There were errors during jobs deletion: zowe-jobs:/demo.zosmf/JOB08156
{
  errorDetails: 'There were errors during jobs deletion: zowe-jobs:/demo.zosmf/JOB08156',
  apiType: 'jes',
  profile: undefined
}
```

**To Reproduce**

1. In the jobs tree, right-click on a profile and select `Show as table`. 
2. Select a job and hit the `Delete` button.
3. Hit `Delete` on the prompt that appears.
4. Get another prompt asking for confirmation that I want to delete the job.
5. Hit `Delete` again and you will receive an error notification.

**Expected behavior**

The prompt should only appear once.

**Screenshots**

<!-- If applicable, add screenshots to help explain your problem. -->

**Desktop (please complete the following information):**

- OS: 
- Zowe Explorer Version: 3.2.0 Snapshot
- (Optional) Zowe CLI Version:
- (Optional) Do you have secure credentials enabled?
- (Optional) Are you using a custom credential manager?

**Additional context**

<!-- Add any other context about the problem here. -->

--- a/packages/zowe-explorer-api/__mocks__/vscode.ts
+++ b/packages/zowe-explorer-api/__mocks__/vscode.ts
@@ -486,17 +486,9 @@ export namespace window {
         return undefined;
     }
 
-    export function createWebviewPanel(
-        viewType: string,
-        title: string,
-        showOptions: ViewColumn | { preserveFocus: boolean; viewColumn: ViewColumn },
-        options?: WebviewPanelOptions & WebviewOptions
-    ): WebviewPanel {
-        return undefined as any;
-    }
-
     const { window: mockWindow } = require("jest-mock-vscode").createVSCodeMock(jest);
     export const showQuickPick = mockWindow.showQuickPick;
+    export const createWebviewPanel = mockWindow.createWebviewPanel;
 
     /**
      * Options to configure the behavior of the message.
@@ -549,14 +541,7 @@ export namespace commands {
         return undefined;
     }
 }
-export class Disposable {
-    /**
-     * Creates a new Disposable calling the provided function
-     * on dispose.
-     * @param callOnDispose Function that disposes something.
-     */
-    constructor() {}
-}
+export const { Disposable } = require("jest-mock-vscode").createVSCodeMock(jest);
 
 export function RelativePattern(_base: string, _pattern: string): {} {
     return {};

--- a/packages/zowe-explorer-api/__tests__/__unit__/vscode/ui/TableView.unit.test.ts
+++ b/packages/zowe-explorer-api/__tests__/__unit__/vscode/ui/TableView.unit.test.ts
@@ -675,8 +675,8 @@ describe("Table.Instance", () => {
                     { a: 3, b: 2, c: 1, d: true, e: 6 },
                 ])
                 .addColumns([{ field: "a" }, { field: "b" }, { field: "c" }, { field: "d" }, { field: "e" }]);
+            const disposeMock = jest.spyOn((WebView as any).prototype, "dispose").mockImplementationOnce(jest.fn());
             const instance = builder.build();
-            const disposeMock = jest.spyOn((WebView as any).prototype, "dispose");
             instance.dispose();
             expect(disposeMock).toHaveBeenCalled();
         });

--- a/packages/zowe-explorer-api/__tests__/__unit__/vscode/ui/WebView.unit.test.ts
+++ b/packages/zowe-explorer-api/__tests__/__unit__/vscode/ui/WebView.unit.test.ts
@@ -44,13 +44,18 @@ describe("WebView unit tests", () => {
         const createWebviewPanelSpy = jest.spyOn(vscode.window, "createWebviewPanel");
         const renderSpy = jest.spyOn(Mustache, "render");
 
+        const disposeMock = jest.fn();
+        const disposable = new vscode.Disposable(disposeMock);
+
         const testView = new WebView("Test Webview Title", "example-folder", { extensionPath: "test/path" } as vscode.ExtensionContext, {
             onDidReceiveMessage: async (_message: any) => {},
         });
+        (testView as any).disposables = [disposable];
         expect(createWebviewPanelSpy).toHaveBeenCalled();
         expect(renderSpy).toHaveBeenCalled();
         (testView as any).dispose();
         expect(testView.panel).toBeUndefined();
+        expect(disposeMock).toHaveBeenCalledTimes(1);
     });
 
     it("returns HTML content from WebView", () => {

--- a/packages/zowe-explorer-api/src/vscode/ui/WebView.ts
+++ b/packages/zowe-explorer-api/src/vscode/ui/WebView.ts
@@ -111,8 +111,9 @@ export class WebView {
                 title: this.title,
             });
             this.webviewContent = builtHtml;
-            if (opts?.onDidReceiveMessage) {
-                this.panel.webview.onDidReceiveMessage(async (message) => opts.onDidReceiveMessage(message));
+            if (opts?.onDidReceiveMessage && !this.eventsRegistered) {
+                this.disposables.push(this.panel.webview.onDidReceiveMessage(async (message) => opts.onDidReceiveMessage(message)));
+                this.eventsRegistered = true;
             }
             this.panel.onDidDispose(() => this.dispose(), null, this.disposables);
             this.panel.webview.html = this.webviewContent;
@@ -143,7 +144,7 @@ export class WebView {
         });
         this.webviewContent = builtHtml;
         if (this.webviewOpts?.onDidReceiveMessage && !this.eventsRegistered) {
-            webviewView.webview.onDidReceiveMessage(async (message) => this.webviewOpts.onDidReceiveMessage(message));
+            this.disposables.push(webviewView.webview.onDidReceiveMessage(async (message) => this.webviewOpts.onDidReceiveMessage(message)));
             this.eventsRegistered = true;
         }
         webviewView.onDidDispose(() => this.dispose(), null, this.disposables);

--- a/packages/zowe-explorer/l10n/poeditor.json
+++ b/packages/zowe-explorer/l10n/poeditor.json
@@ -215,7 +215,7 @@
   "configuration.title": {
     "Zowe Explorer": ""
   },
-  "zowe.ds.paginate.datasetsPerPage": {
+  "zowe.ds.paginate.dataSetsPerPage": {
     "Maximum number of data sets and PDS members to list per page": ""
   },
   "zowe.ds.default.binary": {

--- a/packages/zowe-explorer/l10n/poeditor.json
+++ b/packages/zowe-explorer/l10n/poeditor.json
@@ -216,7 +216,7 @@
     "Zowe Explorer": ""
   },
   "zowe.ds.paginate.dataSetsPerPage": {
-    "Maximum number of data sets and PDS members to list per page": ""
+    "Maximum number of data sets and PDS members to list per page. 0 disables pagination.": ""
   },
   "zowe.ds.default.binary": {
     "Default values of Binary data set creation": ""
@@ -520,6 +520,9 @@
   },
   "zowe.settings.overrideWithEnvironmentVariables": {
     "Indicates if environment variables should override values stored in Zowe configuration profiles on disk.": ""
+  },
+  "zowe.settings.socketConnectTimeout": {
+    "The amount of time to wait for a connection to be initiated with the server, in milliseconds.": ""
   },
   "Enter command here...": "",
   "Page Size:": "",

--- a/packages/zowe-explorer/package.json
+++ b/packages/zowe-explorer/package.json
@@ -1658,10 +1658,10 @@
       "type": "object",
       "title": "%configuration.title%",
       "properties": {
-        "zowe.ds.paginate.datasetsPerPage": {
+        "zowe.ds.paginate.dataSetsPerPage": {
           "default": 100,
           "type": "integer",
-          "description": "%zowe.ds.paginate.datasetsPerPage%",
+          "description": "%zowe.ds.paginate.dataSetsPerPage%",
           "scope": "window"
         },
         "zowe.ds.default.binary": {

--- a/packages/zowe-explorer/package.nls.json
+++ b/packages/zowe-explorer/package.nls.json
@@ -71,7 +71,7 @@
   "cancelJobs": "Cancel Job",
   "getJobJcl": "Get JCL",
   "configuration.title": "Zowe Explorer",
-  "zowe.ds.paginate.datasetsPerPage": "Maximum number of data sets and PDS members to list per page. 0 disables pagination.",
+  "zowe.ds.paginate.dataSetsPerPage": "Maximum number of data sets and PDS members to list per page. 0 disables pagination.",
   "zowe.ds.default.binary": "Default values of Binary data set creation",
   "zowe.ds.default.c": "Default values of C data set creation",
   "zowe.ds.default.classic": "Default values of Classic data set creation",

--- a/packages/zowe-explorer/src/configuration/Constants.ts
+++ b/packages/zowe-explorer/src/configuration/Constants.ts
@@ -79,7 +79,7 @@ export class Constants {
     public static readonly SETTINGS_AUTOMATIC_PROFILE_VALIDATION = "zowe.automaticProfileValidation";
     public static readonly SETTINGS_SECURE_CREDENTIALS_ENABLED = "zowe.security.secureCredentialsEnabled";
     public static readonly SETTINGS_CHECK_FOR_CUSTOM_CREDENTIAL_MANAGERS = "zowe.security.checkForCustomCredentialManagers";
-    public static readonly SETTINGS_DATASETS_PER_PAGE = "zowe.ds.paginate.datasetsPerPage";
+    public static readonly SETTINGS_DATASETS_PER_PAGE = "zowe.ds.paginate.dataSetsPerPage";
     public static readonly LOGGER_SETTINGS = "zowe.logger";
     public static readonly SETTINGS_OVERRIDE_WITH_ENV_VAR = "zowe.settings.overrideWithEnvironmentVariables";
     public static EXTENDER_CONFIG: imperative.ICommandProfileTypeConfiguration[] = [];

